pimsviewer/viewer.py

```diff
@@ -708,8 +720,15 @@ def to_clipboard(self):
 
     @staticmethod
     @memoize
-    def _get_all_files_in_dir(directory):
-        return sorted([f for f in listdir(directory) if isfile(join(directory, f))], key=natural_keys)
+    def _get_all_files_in_dir(directory, extensions=None):
+        if extensions is None:
+            return sorted(
+                [f for f in listdir(directory) if isfile(join(directory, f))],
+                key=natural_keys)
+        else:
+            return sorted([f for f in listdir(directory) if
+                           isfile(join(directory, f)) and drop_dot(os.path.splitext(f)[1]) in extensions],
+                          key=natural_keys)
```

Comment:
Can you remove the duplicates from the to cases of the if statements? Maybe use a variable `file_list` = [f for ....]  and then call the `return sorted(...)` outside of the `if`. Would benefit readability.